### PR TITLE
fix: fix fetching rc versions of vault cli

### PR DESCRIPTION
**Pull request by matifali**
Fixes where the install script was fetching the RC version

### CODE REVIEW

**Review by mtojek — COMMENTED** [review]
Don't you want to use [index.json](https://releases.hashicorp.com/vault/index.json) endpoint?

**Comment by mtojek** [comment]
Interesting. Can you shed more light on why? I thought that `jq` format was more human-friendly.

**Comment by matifali** [comment]
This is a module, and if a workspace does not have `jq`, then it will fail. Ideally, we want 0 dependencies if possible, and the module should be useable in any workspace.

@mafredri once suggested having `coder util jq|curl|wget|zip|tar` type of commands so that we do not depend on anything.

**Comment by mafredri** [comment]
> This is a module, and if a workspace does not have `jq`, then it will fail. Ideally, we want 0 dependencies if possible, and the module should be useable in any workspace.
> 
> @mafredri once suggested having `coder util jq|curl|wget|zip|tar` type of commands so that we do not depend on anything.

Yup, this is the way, let's reduce dependencies wherever we can 💪🏻. The `coder util` is a beautiful idea but we also want to make sure we don't bloat the agent binary. Most of this can be done in a very basic way using the stdlib though, so shouldn't add much weight.

Then again, maybe we don't need to bundle it in `coder util`, perhaps we can have module dependencies and we have one that's called `coder/modules/util` that contains a few basic commands, or the ability to download them for the relevant platform.

File: vault-token/run.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ unzip_safe() {
 install() {
   # Fetch the latest version of Vault if INSTALL_VERSION is 'latest'
   if [ "$${INSTALL_VERSION}" = "latest" ]; then
-    LATEST_VERSION=$(curl -s https://releases.hashicorp.com/vault/ | grep -oP 'vault/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+    LATEST_VERSION=$(curl -s https://releases.hashicorp.com/vault/ | grep -v 'rc' | grep -oP 'vault/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
```
Comment (mafredri) [inline vault-token/run.sh:35]:
```suggestion
    LATEST_VERSION=$(curl -s https://releases.hashicorp.com/vault/ | grep -v '-rc' | grep -oP 'vault/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
```

**Review by mafredri — APPROVED** [review]

**Comment by matifali** [comment]
@mafredri @mtojek another option is to use
`curl -s https://releases.hashicorp.com/vault/index.json | grep -o '"[0-9]\+\.[0-9]\+\.[0-9]\+"' | tr -d '"' | grep -v 'rc' | sort -V | tail -n 1`
without using `jq` it also uses the index.json endpoint

**Comment by mtojek** [comment]
> another option is to use

I think I like the first option, curl + regular endpoint + grep. As `index.json` is one-line response, I would parse it without `jq`.

**Pull request merged**